### PR TITLE
Muon - Widget; Home Tab Grouping Widget

### PR DESCRIPTION
**Pull request by TWJubb**
Refs #23642 

Requires all core code to be merged.

This PR implements the grouping widget used in the home tab of the new muon interface.

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/Muon/GUI/Common/home_grouping_widget/home_grouping_widget_presenter.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        self._view.set_subtracted_periods(",".join([str(p) for p in subtracted_periods]))
+
+    def handle_periods_changed(self):
+        summed = self.string_to_list(self._view.get_summed_periods())
+        subtracted = self.string_to_list(self._view.get_subtracted_periods())
+
+        subtracted = [i for i in subtracted if i not in summed]
+
+        n_periods = self._model.number_of_periods()
+        bad_periods = [p for p in summed if (p > n_periods) or p == 0] + [p for p in subtracted if
```
Comment (AnthonyLim23) [inline scripts/Muon/GUI/Common/home_grouping_widget/home_grouping_widget_presenter.py]: can we have some better variable names please?

File: scripts/Muon/GUI/Common/home_grouping_widget/home_grouping_widget_view.py [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+
+    def __init__(self, parent=None):
+        super(HomeGroupingWidgetView, self).__init__(parent)
+        self.setup_interface()
+
+        self.alpha_hidden(True)
+        self.periods_hidden(False)
+
+        self._button = QtGui.QMessageBox.Ok
+        self._message_box = QtGui.QMessageBox("Error in substitution", "ERROR !", QtGui.QMessageBox.Warning,
```
Comment (AnthonyLim23) [inline scripts/Muon/GUI/Common/home_grouping_widget/home_grouping_widget_view.py]: is there a reason for not using the muon one?

**Comment by AnthonyLim23** [comment]
is this supposed to have no tests?

**Comment by Mantid-Matthew** [comment]
There are no tests in this PR becuase it lacks some of the other files required to test it properly. The relevant tests have been added in #23657

**Review by AnthonyLim23 — APPROVED** [review]

**Comment by martyngigg** [comment]
The build failure is unrelated.

**Pull request merged**